### DISCUSSION TITLE
Error code 1129 - User ID was not provided

### DISCUSSION THREAD

**Discussion by dbocruzin** [mmohades/Venmo#44] [Q&A] [answered]
Hi there,

I was working on trying to set up a payment request through the API and I'm getting a 400 error (copied below).  I'm able to connect and search for users but I get an error on trying to send a request for money? Any suggestions are greatly appreciated!!

code
venmo.payment.request_money(10.1,'test from API','SomeRanNm123')

Error Code
HttpCodeError: HTTP Status code is invalid. Could not make the request because -> 400 Bad Request.
Error: {'error': {'message': 'User ID was not provided. ', 'code': 1129, 'links': None, 'title': 'Error'}}

**Comment by mmohades** [comment, marked as answer]
The string you've provided is an username, you'll need their user ID to request money. You can search for the user, or use `client.user.get_user_by_username("the username of the user you're looking for")`. This method may return `None` if it can't find the user.
Read more here: https://github.com/mmohades/Venmo/blob/master/venmo_api/apis/user_api.py#L92

**Reply by dbocruzin** [reply]
Thank you! I had been scanning through the files but overlooked the obvious!!
